src/tinyformat.h

```diff
@@ -1052,13 +1053,24 @@ TINYFORMAT_FOREACH_ARGNUM(TINYFORMAT_MAKE_FORMAT_FUNCS)
 
 #endif
 
-// Added for Bitcoin Core
-template<typename... Args>
-std::string format(const std::string &fmt, const Args&... args)
+// Bitcoin Core specific patches
+// Underlying tinyformat is up to date as of upstream commit
+// https://raw.githubusercontent.com/c42f/tinyformat/689695cf58700e6defe3741829564cd682d5ae57/tinyformat.h
+template <typename... Args>
+std::string format(const std::string& fmt, const Args&... args)
 {
-    std::ostringstream oss;
-    format(oss, fmt.c_str(), args...);
-    return oss.str();
+    return format(fmt.c_str(), args...);
+}
+
+template <typename... Args>
+std::string format_no_throw(const std::string& fmt, const Args&... args)
```

Comment:
How can I (or the compiler) be sure that no other exceptions (e.g. `std::out_of_range`) are thrown?

Reply:
There is really no shortcut here AFAIK: reading + reasoning would be required :-)

Do you see any places where `std::out_of_range` could be thrown that is reachable from `format`?

Reply:
I think the point of adding "noexcept" isn't so much because you know a function won't throw; it's that you know that exceptions thrown anyway can be regarded as fatal errors.

Reply:
@sipa Stroustrup and Sutter:

> [Use noexcept when exiting a function because of a throw is impossible or unacceptable.](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#e12-use-noexcept-when-exiting-a-function-because-of-a-throw-is-impossible-or-unacceptable)

Reply:
I see, so it is equivalent to `std::terminate`. I think I will leave that for later, as I am not sure that immediate termination is better than offering the gui or daemon to catch the exception

Reply:
I think it is unwise to give the caller the impression that the function cannot throw by calling it `format_no_throw` if we're not willing to back that with a `noexcept` guarantee.

If it is potentially throwing it shouldn't be called `no_throw` :-)

Reply:
Maybe add a comment like `This function wraps format() to return error messages instead of throwing format_error. All other format() exceptions are thrown away.`

Reply:
Added a comment